crates/edr_evm/src/transaction.rs

```diff
@@ -98,14 +98,20 @@ pub fn validate<TransactionT: Transaction>(
 }
 
 /// Calculates the initial cost of a transaction.
-pub fn initial_cost(transaction: &impl Transaction, spec_id: SpecId) -> u64 {
-    validate_initial_tx_gas(
+pub fn initial_cost(transaction: &transaction::Signed, spec_id: SpecId) -> u64 {
+    let authorization_list_num = transaction
+        .authorization_list()
+        .map(AuthorizationList::len)
+        .unwrap_or_default() as u64;
```

Comment:
Nit: would read nicer with a comment "usize is guaranteed to fit into u64"

Reply:
Addressed in https://github.com/NomicFoundation/edr/pull/791/commits/86ccde1007aff43060f0f24e3112d61e5b8a7f5a